Critical error running junit4 hooks

When running `repobee repos clone` with the `junit` and `javac` hooks I get the following error:

```
Executing post_clone hooks:  47%|███████▍        | 7/15 [00:04<00:05,  1.52it/s][ERROR] critical
Traceback (most recent call last):
  File "/Users/janneschyffert/.repobee/env/lib/python3.10/site-packages/repobee_junit4/junit4.py", line 132, in post_clone
    compile_succeeded, compile_failed = self._compile_all(repo)
  File "/Users/janneschyffert/.repobee/env/lib/python3.10/site-packages/repobee_junit4/junit4.py", line 179, in _compile_all
    compile_succeeded, compile_failed = _java.pairwise_compile(
  File "/Users/janneschyffert/.repobee/env/lib/python3.10/site-packages/repobee_junit4/_java.py", line 194, in pairwise_compile
    status, msg, prod_class_path = _pairwise_compile(
  File "/Users/janneschyffert/.repobee/env/lib/python3.10/site-packages/repobee_junit4/_java.py", line 275, in _pairwise_compile
    potential_prod_classes = _get_matching_prod_classes(
  File "/Users/janneschyffert/.repobee/env/lib/python3.10/site-packages/repobee_junit4/_java.py", line 310, in _get_matching_prod_classes
    return [
  File "/Users/janneschyffert/.repobee/env/lib/python3.10/site-packages/repobee_junit4/_java.py", line 313, in <listcomp>
    if file.name == prod_class_name and extract_package(file) == package
  File "/Users/janneschyffert/.repobee/env/lib/python3.10/site-packages/repobee_junit4/_java.py", line 84, in extract_package
    with class_.open(encoding=sys.getdefaultencoding(), mode="r") as file:
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/pathlib.py", line 1117, in open
    return self._accessor.open(self, mode, buffering, encoding, errors,
IsADirectoryError: [Errno 21] Is a directory: '/var/folders/9k/5mkds41d7qvcd95xrb72xbt40000gn/T/tmpa4_0e6lo/947c16c3f94945663c0b4c0fa20f9a8929fb8cf9/src/Indamon.java'
Executing post_clone hooks: 100%|███████████████| 15/15 [00:08<00:00,  1.68it/s]
```

The hooks run regardless and seem to be working correctly despite the error. This is my repobee-config:
```
[repobee]
user = jannesc
token = xxxxxxxxxx
students_file = /Users/janneschyffert/Documents/INDA-TA/students.txt

[junit4]
junit4_reference_tests_dir = /Users/janneschyffert/Documents/INDA-TA/test-files
junit4_hamcrest_path = /Users/janneschyffert/Documents/INDA-TA/test-jars/hamcrest-core-1.3.jar
junit4_junit_path = /Users/janneschyffert/Documents/INDA-TA/test-jars/junit-4.13.1.jar
junit4_timeout = 5
```

--- a/.flake8
+++ b/.flake8
@@ -1,0 +1,2 @@
+[flake8]
+extend-ignore = E501

--- a/repobee_junit4/_junit4_runner.py
+++ b/repobee_junit4/_junit4_runner.py
@@ -14,8 +14,8 @@ from repobee_junit4 import _output
 
 
 LOGGER = daiquiri.getLogger(__file__)
-HAMCREST_JAR_PATTERN = fr"([^{os.pathsep}]*hamcrest-core-1.3.jar)"
-JUNIT4_JAR_PATTERN = fr"([^{os.pathsep}]*junit-4\.\d+\.(\d+\.)?jar)"
+HAMCREST_JAR_PATTERN = rf"([^{os.pathsep}]*hamcrest-core-1.3.jar)"
+JUNIT4_JAR_PATTERN = rf"([^{os.pathsep}]*junit-4\.\d+\.(\d+\.)?jar)"
 
 _DEFAULT_SECURITY_POLICY_TEMPLATE = """grant {{
 }};

--- a/repobee_junit4/junit4.py
+++ b/repobee_junit4/junit4.py
@@ -168,7 +168,9 @@ class JUnit4Hooks(plug.Plugin, plug.cli.CommandExtension):
             a tuple of lists ``(succeeded, failed)``, where ``succeeded``
             are tuples on the form ``(test_class, prod_class)`` paths.
         """
-        java_files = list(repo.path.rglob("*.java"))
+        java_files = [
+            file for file in repo.path.rglob("*.java") if file.is_file()
+        ]
         assignment_name = self._extract_assignment_name(repo.name)
         reference_test_classes = self._find_test_classes(assignment_name)
         test_classes = (

--- a/tests/integration_tests/repos/directory-with-java-suffix-week-10/README.md
+++ b/tests/integration_tests/repos/directory-with-java-suffix-week-10/README.md
@@ -1,0 +1,1 @@
+This repo should pass all tests

--- a/tests/integration_tests/repos/directory-with-java-suffix-week-10/src/Fibo.java/Fibo.java
+++ b/tests/integration_tests/repos/directory-with-java-suffix-week-10/src/Fibo.java/Fibo.java
@@ -1,0 +1,23 @@
+/**
+ * Class for calculating Fibonacci numbers.
+ */
+
+public class Fibo {
+    private long prev;
+    private long current;
+
+    public Fibo() {
+        prev = 0;
+        current = 1;
+    }
+
+    /**
+     * Generate the next Fibonacci number.
+     */
+    public long next() {
+        long ret = prev;
+        prev = current;
+        current = ret + current;
+        return ret;
+    }
+}

--- a/tests/integration_tests/test_end_to_end.py
+++ b/tests/integration_tests/test_end_to_end.py
@@ -76,6 +76,7 @@ UNAUTHORIZED_NETWORK_ACCESS_REPO = (
 BAD_TESTS_REPO = REPO_DIR / "student-with-bad-tests-week-10"
 DUPLICATE_TESTS_REPO = REPO_DIR / "student-with-duplicate-tests-week-10"
 ENDLESS_WHILE_LOOP = REPO_DIR / "endless-loop-week-10"
+DIRECTORY_WITH_JAVA_SUFFIX_REPO = REPO_DIR / "directory-with-java-suffix-week-10"
 
 assert SUCCESS_REPO.exists(), "test pre-requisite error, dir must exist"
 assert FAIL_REPO.exists(), "test pre-requisite error, dir must exist"
@@ -167,6 +168,16 @@ class TestPostClone:
 
         assert result.status == plug.Status.WARNING
         assert "Student wrote a bad test" in str(result.msg)
+
+    def test_handles_directory_with_java_suffix(self):
+        """Should not attempt to read a directory with .java suffix like a file."""
+        hooks = setup_hooks(verbose=True)
+
+        result = hooks.post_clone(
+            wrap_in_student_repo(DIRECTORY_WITH_JAVA_SUFFIX_REPO), api=None
+        )
+
+        assert result.status == plug.Status.SUCCESS
 
     def test_handles_duplicate_student_tests(self):
         hooks = setup_hooks(run_student_tests=True, verbose=True)

--- a/tests/integration_tests/test_end_to_end.py
+++ b/tests/integration_tests/test_end_to_end.py
@@ -76,7 +76,9 @@ UNAUTHORIZED_NETWORK_ACCESS_REPO = (
 BAD_TESTS_REPO = REPO_DIR / "student-with-bad-tests-week-10"
 DUPLICATE_TESTS_REPO = REPO_DIR / "student-with-duplicate-tests-week-10"
 ENDLESS_WHILE_LOOP = REPO_DIR / "endless-loop-week-10"
-DIRECTORY_WITH_JAVA_SUFFIX_REPO = REPO_DIR / "directory-with-java-suffix-week-10"
+DIRECTORY_WITH_JAVA_SUFFIX_REPO = (
+    REPO_DIR / "directory-with-java-suffix-week-10"
+)
 
 assert SUCCESS_REPO.exists(), "test pre-requisite error, dir must exist"
 assert FAIL_REPO.exists(), "test pre-requisite error, dir must exist"

